client/my-sites/plugins/controller.js

```diff
@@ -186,16 +186,18 @@ function renderProvisionPlugins( context ) {
 }
 
 const controller = {
-	plugins( filter, context, next ) {
+	plugins( context, next ) {
+		context.params.pluginFilter = context.params.pluginFilter || 'all';
 		const siteUrl = route.getSiteFragment( context.path );
-		const basePath = route.sectionify( context.path ).replace( '/' + filter, '' );
+		const basePath = route
+			.sectionify( context.path )
+			.replace( '/' + context.params.pluginFilter, '' );
 
 		// bail if no site is selected and the user has no Jetpack sites.
 		if ( ! siteUrl && ! hasJetpackSites( context.store.getState() ) ) {
 			return next();
 		}
 
-		context.params.pluginFilter = filter;
```

Comment:
Oh wow. The code that keeps on giving :roll_eyes:

Reply:
So let's restore this ugly piece of code here and go with https://github.com/Automattic/wp-calypso/pull/20019/files#r152092561.